doc/ufunc_examples.rst

```diff
@@ -0,0 +1,317 @@
+shift the data's position along that axis.
+Therefore our signature should look something like this ``"(X:left,Y:center),(X:center,Y:left)->(X:center,Y:center)"``.
+
+We also need to pad the data to replace the elements that will be removed by the `diff` function, so
+our grid ufunc can be defined like this
+
+.. ipython:: python
+
+    @as_grid_ufunc(
+        "(X:left,Y:center),(X:center,Y:left)->(X:center,Y:center)",
+        boundary_width={"X": (0, 1), "Y": (0, 1)},
+    )
+    def divergence(u, v):
+        u_diff_x = diff(u, axis=-2)
+        v_diff_y = diff(v, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        div = u_diff_x[..., :-1] + v_diff_y[..., :-1, :]
+        return div
+
+Here we have treated the components of the ``(U, V)`` vector as independent scalars.
+
+Now we can compute the divergence of our example vector field
+
+.. ipython:: python
+
+    div = divergence(grid, ds["U"], ds["V"], axis=[("X", "Y"), ("X", "Y")])
+
+We can see the result lies on the expected coordinate positions
+
+.. ipython:: python
+
+    div.coords
+
+and the resulting divergence looks like it corresponds with the arrows of the vector field above
+
+.. ipython:: python
+
+    import matplotlib.pyplot as plt
+
+    div.plot(x="x_c", y="y_c")
+    colocated.plot.quiver("x_c", "y_c", u="U", v="V")
+
+    @savefig div_vector_field.png width=4in
+    plt.gcf()
+
+
+
+Gradient
+~~~~~~~~
+
+The gradient is almost like the opposite of divergence in the sense that it accepts one scalar and returns multiple vectors.
+
+Let's first define a tracer field ``T``, which we imagine will start off localised in the center of the domain.
+
+.. ipython:: python
+
+    def gaussian(x_coord, y_coord, x_pos, y_pos, A, w):
+        return A * np.exp(
+            -0.5 * ((x_coord - x_pos) ** 2 + (y_coord - y_pos) ** 2) / w ** 2
+        )
+
+
+    ds["T"] = gaussian(grid_ds.x_c, grid_ds.y_c, x_pos=7.5, y_pos=7.5, A=50, w=2)
+
+    @savefig tracer_field.png width=4in
+    ds["T"].plot.contourf(x="x_c", vmax=60)
+
+Computing the first-order gradient will again move the data onto different grid positions,
+so the signature for a gradient ufunc will need to reflect this
+and our definition is similar to the derivative case.
+
+.. ipython:: python
+
+    def gradient(a):
+        a_diff_x = diff(a, axis=-2)
+        a_diff_y = diff(a, axis=-1)
+        # Need to trim off elements so that the two arrays have same shape
+        return a_diff_x[..., :-1], a_diff_y[..., :-1, :]
+
+Now we can compute the gradient of our example scalar field
+
+.. ipython:: python
+
+    ds["grad_T_x"], ds["grad_T_y"] = grid.apply_as_grid_ufunc(
+        gradient,
+        ds["T"],
+        axis=[("X", "Y")],
+        signature="(X:center,Y:center)->(X:left,Y:center),(X:center,Y:left)",
+        boundary_width={"X": (1, 0), "Y": (1, 0)},
+    )
+
+Again in order to plot this as a vector field we should first interpolate it
```

Comment:
I think one of those 'note' boxes here would be handy, where you point to the fact that you are now using `grid.apply_as_grid_ufunc` and not the decorator, but that both are equivalent.